src/mlpack/methods/ann/layer/add_merge_impl.hpp

```diff
@@ -150,19 +150,17 @@ template<typename InputDataType, typename OutputDataType,
          typename... CustomLayers>
 template<typename Archive>
 void AddMerge<InputDataType, OutputDataType, CustomLayers...>::serialize(
-    Archive& ar, const unsigned int version)
+    Archive& ar, const uint32_t /* version */)
 {
   // Be sure to clear other layers before loading.
-  if (Archive::is_loading::value)
+  if (cereal::is_loading<Archive>())
     network.clear();
 
-  ar & BOOST_SERIALIZATION_NVP(network);
-  ar & BOOST_SERIALIZATION_NVP(model);
-  ar & BOOST_SERIALIZATION_NVP(run);
-
-  if (version >= 1)
-    ar & BOOST_SERIALIZATION_NVP(ownsLayers);
-  else if (Archive::is_loading::value)
+  ar(CEREAL_VECTOR_VARIANT_POINTER(network));
+  ar(CEREAL_NVP(model));
+  ar(CEREAL_NVP(run));
+  ar(CEREAL_NVP(ownsLayers));
+  if (cereal::is_loading<Archive>())
     ownsLayers = !model;
```

Comment:
We can remove this---take a look at the original logic: if `version` was greater than or equal to 1, we serialized `ownsLayers`; *otherwise*, if we were loading, we set `ownsLayers` to `!model`.  But now, we only have to consider the case where the version is the newest (since we are removing reverse compatibility) and thus we don't need the `if` anymore. :+1: